### PR TITLE
create getRecommendedSpaces endpoint

### PR DESCRIPTION
**Pull request by furkanakkurt1335**
I've created the endpoint for getting recommended spaces based on somebody's interests and inferred interests from datamuse's API.

### CODE REVIEW

File: app/server/src/routes/space/space.controller.js [hunk 1]
```diff
@@ -165,6 +167,42 @@ const SpaceController = {
...
+      var space_ids = [];
+      for (inf_in of inferred_interests) {
+        interests.push(inf_in.word);
+      }
+      for (interest_t of interests) {
+        let spaces_t = await SpaceModel.Space.find({$text: {$search: `\"${interest_t}\"`}})
+                              .limit(2)
+                              .exec();
+        for (space_t of spaces_t) {
+          if (!space_ids.includes(space_t._id.toString())) {
```
Comment (ersoykadir) [inline app/server/src/routes/space/space.controller.js:195]: burda bir de daha önce enroll olduğu spaceleri exclude etmek için check yapalım

File: app/server/src/routes/space/space.controller.js [hunk 2]
```diff
@@ -165,6 +167,42 @@ const SpaceController = {
...
+        inferred_interests = inferred_interests.concat(trg_result.data);
+      }
+      inferred_interests.sort((a, b) => b.score - a.score);
+      var spaces = [];
+      var space_ids = [];
+      for (inf_in of inferred_interests) {
+        interests.push(inf_in.word);
+      }
+      for (interest_t of interests) {
+        let spaces_t = await SpaceModel.Space.find({$text: {$search: `\"${interest_t}\"`}})
```
Comment (ersoykadir) [inline app/server/src/routes/space/space.controller.js:191]: spacelerin infolarını searchlemeye ek olarak tagleri de eklesek iyi olabilir

File: app/server/src/models/space/space.model.js [hunk 3]
```diff
@@ -75,6 +75,7 @@ const spaceSchema = new mongoose.Schema(
   },
   { timestamps: true }
 );
+spaceSchema.index({info: 'text'});
```
Comment (ersoykadir) [inline app/server/src/models/space/space.model.js:78]: buraya, tags:'text' ekleyince ikisini de arıyor olabilir, net araştırıp emin olursan iyi olur

**Review by ersoykadir — COMMENTED** [review]
We probably need to try this after changes, if it does not produce enough matches with our method of searching using related words, we may have to try extra stuff.

File: app/server/src/routes/space/space.controller.js [hunk 4]
```diff
@@ -165,6 +167,42 @@ const SpaceController = {
...
+    try {
+      const user_id = req.auth.id;
+      const user = await UserModel.User.findById(user_id);
+      const personalInfo = await PersonalInfoModel.PersonalInfo.findOne({_id: user.personal_info});
+      var interests = personalInfo.interests;
+      const url = "https://api.datamuse.com/words?";
+      var inferred_interests = [];
+      for (interest_t of interests) {
+        if (interest_t.includes(" ")) interest_t = interest_t.replace(" ", "+");
+        const trg_url = `${url}max=5&ml=${interest_t}`;
```
[the remark this answers is not in the document]
Comment (furkanakkurt1335) [reply app/server/src/routes/space/space.controller.js]: I increased it to 10, 1 response took 2078.914 ms. It's fine I think.

File: app/server/src/routes/space/space.controller.js (hunk 1 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/space/space.controller.js:195]: done!

File: app/server/src/models/space/space.model.js (hunk 3 above)
Comment (furkanakkurt1335) [reply app/server/src/models/space/space.model.js]: done!

File: app/server/src/routes/space/space.controller.js (hunk 2 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/space/space.controller.js:191]: done!

**Comment by furkanakkurt1335** [comment]
I tested locally, it produces enough results I think.

**Review by ersoykadir — APPROVED** [review]
Looks good, we should increase our space data and try it out

**Pull request merged**